# Day 30: Working with Exception Handling and JSON

## Reflection  
  Today, I continued my journey in the 100 Days of Python challenge by completing Day 30. It was an informative and productive day where I delved deeper into the following concepts:

  1. I revisited the concept of Exception Handling in Python, which I was already familiar with.
  2. I discovered the usefulness of the else statement, which executes when no exceptions occur.
  3. I learned how to raise my own custom exceptions.
  4. I delved into using the JSON library for data manipulation.
  5. I gained knowledge in reading, writing, and updating data in JSON format.
  
  In addition, I successfully tackled two exercises. First, I added error handling to my [NATO Alphabet]https://github.com/johnivanpuayap/NatoAlphabet() project. Second, I significantly improved my [Password Manager](https://github.com/johnivanpuayap/PasswordManager) App by transitioning from a data.txt file to a JSON format. This enhancement involved adding robust exception handling, implementing a search feature for websites, and incorporating message boxes.

  I had an incredibly productive and enriching day of learning. I'm genuinely excited to explore the upcoming Flash Card App Capstone Project in the next phase of the challenge. With 70 days to go, I'm looking forward to making further progress!
## Progress
 - Completed Day 30 of 100 Days of Python